stdlib/src/utils/string_slice.mojo

```diff
@@ -1108,6 +1291,150 @@ fn _to_string_list[
     return _to_string_list[len_fn, unsafe_ptr_fn](items)
 
 
+@always_inline
+fn _split[
+    T0: Stringlike, //,
+    has_maxsplit: Bool,
+    has_sep: Bool,
+    T1: Stringlike = StringLiteral,
+](src_str: T0, sep: Optional[T1], maxsplit: Int) -> List[String]:
+    @parameter
+    if has_sep:
+        return _to_string_list(
+            _split_impl[has_maxsplit](src_str, sep.value(), maxsplit)
+        )
+    else:
+        return _to_string_list(_split_impl[has_maxsplit](src_str, maxsplit))
+
+
+fn _split_sl[
+    O: ImmutableOrigin, //,
+    has_maxsplit: Bool,
+    has_sep: Bool,
+    T: Stringlike = StringLiteral,
+](src_str: StringSlice[O], sep: Optional[T], maxsplit: Int) -> List[
+    StringSlice[O]
+] as output:
+    var items: List[Span[Byte, O]]
+
+    @parameter
+    if has_sep:
+        items = rebind[__type_of(items)](
+            _split_impl[has_maxsplit](src_str, sep.value(), maxsplit)
+        )
+    else:
+        items = rebind[__type_of(items)](
+            _split_impl[has_maxsplit](src_str, maxsplit)
+        )
+
+    i_len = len(items)
+    out_ptr = UnsafePointer[StringSlice[O]].alloc(i_len)
+    i = 0
+    for item in items:
+        (out_ptr + i).init_pointee_move(StringSlice[O](unsafe_from_utf8=item[]))
+        i += 1
+    output = __type_of(output)(
+        unsafe_pointer=out_ptr, size=i_len, capacity=i_len
+    )
+
+
+fn _split_impl[
+    T0: Stringlike,
+    T1: Stringlike,
+    O: ImmutableOrigin, //,
+    has_maxsplit: Bool,
+](ref [O]src_str: T0, sep: T1, maxsplit: Int) -> List[Span[Byte, O]] as output:
+    sep_len = len(sep.as_bytes[False, O]())
+    if sep_len == 0:
+        iterator = src_str.__iter__[False, __origin_of(src_str)]()
+        i_len = len(iterator) + 2
+        out_ptr = UnsafePointer[Span[Byte, O]].alloc(i_len)
+        out_ptr[0] = src_str.as_bytes[False, O]()[0:0]
+        i = 1
+        for s in iterator:
+            out_ptr[i] = rebind[Span[Byte, O]](s.as_bytes[False, O]())
+            i += 1
+        out_ptr[i] = src_str.as_bytes[False, O]()[-1:-1]
+        output = __type_of(output)(
+            unsafe_pointer=out_ptr, size=i_len, capacity=i_len
+        )
+        return
+
+    alias prealloc = 32  # guessing, Python's implementation uses 12
+    amnt = prealloc
+
+    @parameter
+    if has_maxsplit:
+        amnt = maxsplit + 1 if maxsplit < prealloc else prealloc
+    output = __type_of(output)(capacity=amnt)
+    str_byte_len = len(src_str.as_bytes[False, O]())
+    lhs = 0
+    rhs = 0
+    items = 0
+    ptr = src_str.as_bytes[False, O]().unsafe_ptr()
+    # str_span = src_str.as_bytes[False, O]() # FIXME(#3295)
```

Comment:
**Question** Is that GH issue link right?  #3295 refers to https://github.com/modularml/mojo/issues/3296 which is closed.  Ditto a few lines below in the next FIXME.

Reply:
Sorry, meant to link to issue #3526 and mean solve with #3548